packages/grafana-runtime/src/analytics/plugins/usePluginInteractionReporter.ts

```diff
@@ -0,0 +1,38 @@
+import {
+  type DataSourcePluginContextType,
+  isDataSourcePluginContext,
+  type PluginContextType,
+  usePluginContext,
+} from '@grafana/data';
+
+import { reportInteraction } from '../utils';
+
+import { createDataSourcePluginEventProperties, createPluginEventProperties } from './eventProperties';
+
+export function usePluginInteractionReporter(): typeof reportInteraction {
+  const context = usePluginContext();
+
+  if (isDataSourcePluginContext(context)) {
+    return createDataSourceReporter(context);
+  }
+
+  return createPluginReporter(context);
```

Comment:
this is creating a new reporting on ever render, can't we add this to the context instead (so only create it conce)

Reply:
like a reportInteraction function directly on the context seems like good and simpler idea

Reply:
Yes, that would be an alternative or just wrap it in a memo